### PR TITLE
EZP-27458: Add Field Facet Visitor

### PR DESCRIPTION
**Pull request by wizhippo**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-27458](https://jira.ez.no/browse/EZP-27458)
| **Improvement**| yes
| **New feature**    | no
| **Target version** | master for features
| **BC breaks**      | yes
| **Tests pass**     | yes/no
| **Doc needed**     | yes

This is needed to implement Field Facet Visitors in solr.
We need the change the API so that both solr and elasticsearch can support it.

**TODO**:
- [x] Implement feature / fix a bug.
- [ ] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [ ] Ask for Code Review.

### CODE REVIEW

**Comment by andrerom** [comment]
👍 

@lserwatka proposed API change here for facets, it's not implemented anywhere yet so it's more in order to be able to expand on Field facet support in 2.5. Anyone with elastic knowhow who can review this? _(Solr parts checks out, as this maps straight to Solr concepts: https://github.com/ezsystems/ezplatform-solr-search-engine/pull/129)_

File: eZ/Publish/API/Repository/Values/Content/Query/FacetBuilder/FieldFacetBuilder.php [hunk 1]
```diff
@@ -35,9 +35,32 @@ class FieldFacetBuilder extends FacetBuilder
...
      * @var string
+     *
+     * @deprecated Use FieldFacetBuilder::prefix, FieldFacetBuilder::contains instead
      */
     public $regex;
 
+    /**
+     * A value that field values must start with.
+     *
+     * @var string
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Values/Content/Query/FacetBuilder/FieldFacetBuilder.php]: add a `@since 2.5` on all new properties

**Comment by lserwatka** [comment]
@andrerom we are not going to make it for 2.5. We reached the scope and capacity for this release.

File: eZ/Publish/API/Repository/Values/Content/Query/FacetBuilder/FieldFacetBuilder.php [hunk 2]
```diff
@@ -35,9 +35,38 @@ class FieldFacetBuilder extends FacetBuilder
      * A regex filter for field values.
      *
      * @var string
+     *
+     * @deprecated Use FieldFacetBuilder::prefix, FieldFacetBuilder::contains instead
```
Comment (andrerom) [inline eZ/Publish/API/Repository/Values/Content/Query/FacetBuilder/FieldFacetBuilder.php]: Can skip this change, applied similar message now on 6.7 and up.

**Review by andrerom — APPROVED** [review]

**Pull request closed without merging**